### PR TITLE
Move the logic for create Athena urls to a function

### PR DESCRIPTION
**Pull request by thijs-obj**
Add helper function to construct Athena urls

### CODE REVIEW

File: bach/bach/utils.py [hunk 1]
```diff
@@ -188,3 +190,38 @@ def merge_sql_statements(dialect: Dialect, sql_statements: List[str]) -> List[st
         return sql_statements
     combined = '; '.join(sql_statements)
     return [combined]
+
+
+def athena_construct_engine_url(
+    *,
+    aws_access_key_id: str,
+    aws_secret_access_key: str,
```
Comment (thijs-obj) [inline bach/bach/utils.py:198]: Actually just had a short discussion with Mike. These parameters might not be required. As in some cases (e.g. when running on AWS) it makes more sense to let the library automagically handle this. Will look into this in a bit.

File: bach/bach/utils.py [hunk 2]
```diff
@@ -188,3 +190,56 @@ def merge_sql_statements(dialect: Dialect, sql_statements: List[str]) -> List[st
...
+
+def athena_construct_engine_url(
+    *,
+    aws_access_key_id: str = None,
+    aws_secret_access_key: str = None,
+    region_name: str,
+    schema_name: str,
+    s3_staging_dir: str,
+    athena_work_group: Optional[str] = None,
+    catalog_name: Optional[str] = None
```
Comment (KathiaBarahona) [inline bach/bach/utils.py:203]: maybe `location` param will be nice to have

**Review by KathiaBarahona — APPROVED** [review]

File: bach/bach/utils.py (hunk 2 above)
Comment (thijs-obj) [reply bach/bach/utils.py:203]: I did an experiment with adding `location`, but if I then did `create table` with no location specified (through `DataFrame.database_create_table()`, then it did not write to the location parameter in the connection string, but to the staging_dir. So I think adding `location` doesn't work? Or am I doing it wrong

**Comment by thijs-obj** [comment]
Still a review thread open. But think OK to merge like this, also because https://github.com/objectiv/objectiv.io/pull/596 relies on this. I'll monitor replies here and make follow-up PR if needed.

**Pull request merged**